components/gfx/font_template.rs

```diff
@@ -2,40 +2,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use style::computed_values::font_weight;
+use font::FontHandleMethods;
 use platform::font_context::FontContextHandle;
 use platform::font::FontHandle;
 use platform::font_template::FontTemplateData;
 
+use style::computed_values::{font_stretch, font_weight};
 use sync::{Arc, Weak};
-use font::FontHandleMethods;
 
 /// Describes how to select a font from a given family.
 /// This is very basic at the moment and needs to be
 /// expanded or refactored when we support more of the
 /// font styling parameters.
-#[deriving(Clone)]
+#[deriving(Clone, PartialEq)]
 pub struct FontTemplateDescriptor {
     pub weight: font_weight::T,
+    pub stretch: font_stretch::T,
     pub italic: bool,
 }
 
 impl FontTemplateDescriptor {
-    pub fn new(weight: font_weight::T, italic: bool) -> FontTemplateDescriptor {
+    #[inline]
+    pub fn new(weight: font_weight::T, stretch: font_stretch::T, italic: bool)
+               -> FontTemplateDescriptor {
         FontTemplateDescriptor {
             weight: weight,
+            stretch: stretch,
             italic: italic,
         }
     }
 }
 
-impl PartialEq for FontTemplateDescriptor {
```

Comment:
I don't think you can remove this and replace with PartialEq because the is_bold() method on weight is not an equality check?